soroban-env-host/src/auth.rs

```diff
@@ -796,13 +856,15 @@ impl AuthorizationManager {
         Vec::<Val>::charge_bulk_init_cpy(auth_entries.len() as u64, host)?;
         trackers.reserve(auth_entries.len());
         for e in auth_entries {
-            trackers.push(InvokerContractAuthorizationTracker::new(host, e)?)
+            trackers.push(
+                InvokerContractAuthorizationTracker::new_with_curr_contract_as_invoker(host, e)?,
+            )
         }
         Ok(())
     }
 
     // metering: covered by components
-    fn verify_contract_invoker_auth(
+    fn maybe_authorize_as_invoker_contract(
```

Comment:
Ah, I see, you can read "authorize" here meaning give-authorization rather than check-authorization as I meant it. Fair enough. How about `maybe_check_invoker_contract_auth`? I was tripped up by the notion of "verify" here, thinking that if it returned false or a failure there was something _wrong_; when it is really just a "check one of the possible ways we might be authorized" function, I wanted to convey that there's nothing wrong if this just happens to return false.

Reply:
Yeah, 'check' is good.